ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -377,18 +385,19 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
         );
         retryWatch.start();
         return () => retryWatch.stop();
-    }, [namespace, name]);
+    }, [namespace, name, isWfInDB]);
```

Comment:
Could you check if different cases for workflow deletion still work as before?

Reply:
I meant the workflow delete button:
1. Delete only live workflow
2. Also delete archived workflow (there will be a checkbox)

Reply:
Good point. I'm wrong. 
The checkbox is not shown because `isInCluster=false`.

We should call [this line](https://github.com/toyamagu-2021/argo-workflows/blob/b9e5e986c533571cacfc37ee2352c5e6860aa8dd/ui/src/app/workflows/components/workflow-details/workflow-details.tsx#L359) but this function throws error described in #11371.

I'm thinking about a better way to make 'isInCluster=true'

Reply:
>  I'm thinking about a better way to make 'isInCluster=true'

This would depend on https://github.com/argoproj/argo-workflows/issues/11367#issuecomment-1640880037?

Reply:
Yes, it should depends on a PR.
I'll submit a PR in Friday, or in this weekend.

Reply:
As commented below, I checked this point with/without a checkbox.
- No checkbox -> delete only live workflow ( `Archived` -> `Persisted` )
- With checkbox -> delete live workflow and archived workflow